harmony/harmony.py

```diff
@@ -35,6 +35,19 @@
 ]
 
 
+class ProcessingFailedException(Exception):
```

Comment:
I agree with the principle and waffled on it a little, because there isn't an obvious second exception of this type.  Most of the other errors that could arise are either disconnected from jobs (server errors, auth errors) or happen before a Job ID exists (validation / submission errors).  I ended up punting and figuring we'd create the base class when that second class came up.